ENH: update feather IO for pyarrow 0.17 / Feather V2 (#33422)
pandas-dev/pandas: doc/source/user_guide/io.rst

Feather is designed to faithfully serialize and de-serialize DataFrames, supporting all of the pandas
dtypes, including extension dtypes such as categorical and datetime with tz.

Several caveats:

* The format will NOT write an ``Index``, or ``MultiIndex`` for the
  ``DataFrame`` and will raise an error if a non-default one is provided. You
  can ``.reset_index()`` to store the index or ``.reset_index(drop=True)`` to
  ignore it.
* Duplicate column names and non-string columns names are not supported

> * Non supported types actual Python object types. These will raise a helpful error message

Reviewer: Missing a word here. Maybe

```suggestion
* object-dtype columns are not supported. This will raise with a helpful error message
```

Reviewer 2: for my edification, does this mean that PeriodIndex or Series[Period] is supported?  If so, is that a change from the older version?
